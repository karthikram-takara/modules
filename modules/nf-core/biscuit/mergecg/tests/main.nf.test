nextflow_process {

    name "Test Process BISCUIT_MERGECG"
    script "../main.nf"
    process "BISCUIT_MERGECG"
    tag "modules"
    tag "modules_nfcore"
    tag "biscuit"
    tag "biscuit/mergecg"

    test("sars_cov_bed") {

        setup {
            run("BISCUIT_INDEX") {
                script "../../index/main.nf"
                process {
                    """
                    input[0] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                    """
                }
            }
        }
        when {
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file('https://github.com/nf-core/test-datasets/raw/modules/data/delete_me/biscuit/test-cg.bed.gz', checkIfExists: true)
                        ]

                input[1] = BISCUIT_INDEX.out.index
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.mergecg_bed).match("out_bed") },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }

    }

}
