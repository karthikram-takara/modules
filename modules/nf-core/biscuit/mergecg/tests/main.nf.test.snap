{
    "versions": {
        "content": [
            [
                "versions.yml:md5,caa3e8096399d35c725d7e2d430605c9"
            ]
        ],
        "timestamp": "2023-11-28T21:13:45.569940036"
    },
    "out_bed": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.bed.gz:md5,8136632b0c17c61857a302a57ae302ef"
                ]
            ]
        ],
        "timestamp": "2023-11-28T21:13:45.564582687"
    }
}